cylc/flow/scripts/validate.py

```diff
@@ -37,20 +38,44 @@
     TriggerExpressionError
 )
 import cylc.flow.flags
-from cylc.flow.id_cli import parse_id
+from cylc.flow.id_cli import parse_id_async
 from cylc.flow.loggingutil import disable_timestamps
 from cylc.flow.option_parsers import (
     WORKFLOW_ID_OR_PATH_ARG_DOC,
     CylcOptionParser as COP,
     Options,
     icp_option,
 )
+from cylc.flow.pathutil import (
+    get_source_conf_from_id, is_relative_to, get_cylc_run_dir)
 from cylc.flow.profiler import Profiler
 from cylc.flow.task_proxy import TaskProxy
 from cylc.flow.templatevars import get_template_vars
 from cylc.flow.terminal import cli_function
 
 
+def _revalidate(
+    cfg, flow_file, options, workflow_id, mem_log_func=None
+):
+    # Check if this is a rundir - revalidate not allowed otherwise:
+    if not is_relative_to(flow_file, get_cylc_run_dir()):
+        raise WorkflowConfigError(
+            'Revalidate can only be used on an installed workflow')
+
+    # Get the source config
+    source_cfg = WorkflowConfig(
+        workflow_id,
+        get_source_conf_from_id(workflow_id),
+        options,
+        get_template_vars(options, flow_file),
+        output_fname=options.output,
+        mem_log_func=mem_log_func
+    )
+
+    # Update the in-place config with source config items.
+    cfg.pcfg.sparse.update(source_cfg.pcfg.sparse)
```

Comment:
Why is this function implemented differently for `validate` to other commands e.g. `config`?

Reply:
This PR contains two different, but related things 
- Run Cylc Validate/View/Graph/Config on played workflows, retrieving template variables from database.
- Validate against source.

I should probably add validate against source to all four scripts, but those are not stepping stones to VRO.